tests/regression_tests/unstructured_mesh/source_sampling/test.py

```diff
@@ -0,0 +1,182 @@
+
+        openmc.run(**kwargs)
+
+    def _compare_results(self):
+        # There are 10000 particles and 1000 hexes, this leads to the average
+        # shown below
+        average_in_hex = 10.0
+
+        call(['../../../../scripts/openmc-track-combine', '-o', 'tracks.h5'] +
+                glob.glob('tracks_p*.h5'))
+
+        # loop over the tracks and get data
+        tracks = openmc.Tracks(filepath='tracks.h5')
+        tracks_born = np.empty((len(tracks), 1))
+
+        instances = np.zeros(1000)
+
+        for i in range(0, len(tracks)):
+            tracks_born[i] = tracks[i].particle_tracks[0].states['cell_id'][0]
+            instances[int(tracks_born[i])-1] += 1
+
+        if self.schemes == "file":
+            assert(instances[0] > 0 and instances[1] > 0)
+            assert(instances[0] > instances[1])
+
+            for i in range(2, len(instances)):
+                assert(instances[i] == 0)
+
+        else:
+            assert(np.average(instances) == average_in_hex)
+            assert(np.std(instances) < np.average(instances))
+            assert(np.amax(instances) < np.average(instances)+6*np.std(instances))
+
+
+    def _cleanup(self):
+        super()._cleanup()
+        output = glob.glob('track*.h5')
+        output += glob.glob('tally*.e')
+        for f in output:
+            if os.path.exists(f):
+                os.remove(f)
+
+param_values = (['libmesh', 'moab'], # mesh libraries
+                ['volume', 'file']) # Element weighting schemes
+
+test_cases = []
+for i, (lib, schemes) in enumerate(product(*param_values)):
+    test_cases.append({'library' : lib,
+                       'schemes' : schemes,
+                       'inputs_true' : 'inputs_true{}.dat'.format(i)})
+
+@pytest.mark.parametrize("test_opts", test_cases)
+def test_unstructured_mesh(test_opts):
+
+    openmc.reset_auto_ids()
+
+    # skip the test if the library is not enabled
+    if test_opts['library'] == 'moab' and not openmc.lib._dagmc_enabled():
+        pytest.skip("DAGMC (and MOAB) mesh not enabled in this build.")
+
+    if test_opts['library'] == 'libmesh' and not openmc.lib._libmesh_enabled():
+        pytest.skip("LibMesh is not enabled in this build.")
+
+    ### Materials ###
+    materials = openmc.Materials()
+
+    water_mat = openmc.Material(material_id=3, name="water")
+    water_mat.add_nuclide("H1", 2.0)
+    water_mat.add_nuclide("O16", 1.0)
+    water_mat.set_density("atom/b-cm", 0.07416)
+    materials.append(water_mat)
+
+    materials.export_to_xml()
+
+    ### Geometry ###
+    dimen = 10
+    size_hex = 20.0 / dimen
+
+    ### Geometry ###
+    cell = np.empty((dimen, dimen, dimen), dtype=object)
+    surfaces = np.empty((dimen + 1, 3), dtype=object)
+
+    geometry = openmc.Geometry()
+    universe = openmc.Universe(universe_id=1, name="Contains all hexes")
+
+    for i in range(0,dimen+1):
+        coord = -10.0 + i * size_hex
+        surfaces[i][0] = openmc.XPlane(coord, name="X plane at "+str(coord))

```

Comment:
```suggestion
        surfaces[i][0] = openmc.XPlane(coord, name=f"X plane at {coord}")

```

Same for others like this